### PR TITLE
Scripts/Ragefire Chasm: Implement Adarogg encounter

### LINKED ISSUE #29948: Adarogg boss  Cast Inferno Charge on Aggro have not effect
### Description

Adarogg boss

https://www.youtube.com/watch?v=qFKavyBz7Vw
https://www.wowhead.com/npc=61408/adarogg
https://www.wowhead.com/spell=119405/inferno-charge

 use  this sql  the Inferno Charge only have   the action bar for casting but have not Charge  action 

SET @ENTRY := 61408;
UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid`=@ENTRY;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,0,0,100,0,4000,6000,10000,15000,11,119405,0,0,0,0,0,2,0,0,0,0,0,0,0,'Cast Inferno Charge on Aggro'),
(@ENTRY,0,1,0,0,0,100,0,5000,8000,15000,17000,11,119420,0,0,0,0,0,2,0,0,0,0,0,0,0,'Cast Flame Breath');

how to fix this bugs?  ty

### Expected behaviour

https://www.youtube.com/watch?v=qFKavyBz7Vw

### Steps to reproduce the problem

.go xyz 1805 -4400 -17 1
enter into dungeon
.npc add 61408

### Branch

master

### TC rev. hash/commit

inferno-charge

### Operating system

windows10

### Custom changes

no

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Implement Adarogg encounter in Ragefire Chasm

**Issues addressed:**

Closes #29948

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                adarogg->AI()->DoAction(ACTION_KILL_HOUNDMASTERS);
+
+        return true;
+    }
+};
+
+// 61408 - Adarogg
+struct boss_adarogg : public BossAI
+{
+    using BossAI::BossAI;
```
Comment (mdX7) [inline src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp:91]: drop this line, you gotta explicitly call the ctor of BossAI with `DATA_ADAROGG` as param, if you wouldn't need any specific params you could use the using (like for ScriptedAI, but not here)

File: src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    TaskScheduler scheduler;
```
Comment (mdX7) [inline src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp:214]: `_scheduler`

File: src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    uint8 eatCounter;
```
Comment (mdX7) [inline src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp:215]: `_eatCounter`

File: src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+            corruptedHoundmaster1->AI()->Talk(SAY_HOUNDMASTER_INTRO_0);
+            corruptedHoundmaster2->AI()->Talk(SAY_HOUNDMASTER_INTRO_1);
+
+            scheduler.Schedule(2s, [this](TaskContext task)
+            {
+                DoCast(SPELL_ADJULES_CHOW_TIME);
+                me->GetMotionMaster()->MovePoint(0, IntroPoints[0]);
+
+                task.Schedule(1s, [this](TaskContext task)
```
Comment (mdX7) [inline src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp:130]: movement based on time feels a bit weird here, should this movement happen after `IntroPoint[0]` from above is reached?

**Review by mdX7 — CHANGES_REQUESTED** [review]
`BossBoundaryData` is missing

File: src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp (hunk 4 above)
Comment (aquadeus) [reply src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp:128]: on retail he cast the spell in movement, before reach the point

**Pull request merged**